packages/@aws-cdk/cdk/lib/secret-value.ts

```diff
@@ -0,0 +1,83 @@
+import { DynamicReference, DynamicReferenceService } from './dynamic-reference';
+import { Token } from './token';
+
+/**
+ * Work with secret values in the CDK
+ *
+ * Secret values in the CDK (such as those retrieved from SecretsManager) are
+ * represented as regular strings, just like other values that are only
+ * available at deployment time.
+ *
+ * To help you avoid accidental mistakes which would lead to you putting your
+ * secret values directly into a CloudFormation template, constructs that take
+ * secret values will not allow you to pass in a literal secret value. They do
+ * so by calling `Secret.assertSafeSecret()`.
+ *
+ * You can escape the check by calling `Secret.plainTex()`, but doing
+ * so is highly discouraged.
+ */
+export class SecretValue extends Token {
+  /**
+   * Construct a literal secret value for use with secret-aware constructs
+   *
+   * *Do not use this method for any secrets that you care about.*
+   *
+   * The only reasonable use case for using this method is when you are testing.
+   */
+  public static plainText(secret: string): SecretValue {
+    return new SecretValue(secret);
+  }
+
+  /**
+   * Creates a `SecretValue` with a value which is dynamically loaded from AWS Secrets Manager.
+   * @param secretId The ID or ARN of the secret
+   * @param options Options
+   */
+  public static secretsManager(secretId: string, options: SecretsManagerSecretOptions = { }): SecretValue {
```

Comment:
If you're going to go the way of adding constructor functions here, then you should also add one for SSMPS.

Reply:
Good point.

Reply:
And it should kinda/sorta also be possible to use a NoEcho CfnParameter.

We shouldn't encourage it but it is something people can do today, and SecretsManager *does* cost money per secret per month.

Reply:
Yap adding that too. Enforcing NoEcho is a nice touch.